### PR TITLE
Edits to Client Cfg Guide

### PR DESCRIPTION
**Pull request by Loquacity**
https://github.com/SUSE/spacewalk/issues/7711

Following on from https://github.com/SUSE/doc-susemanager/pull/444 @moio

### CODE REVIEW

**Comment by Loquacity** [comment]
Because the diff is resembling spaghetti:

Screenshot from 2019-05-01 15-57-43

File: modules/client-configuration/pages/client-proxy.adoc [hunk 1]
```diff
@@ -1,15 +1,15 @@
 [[client-proxy]]
 = Registering Clients on a Proxy
 
+Proxy servers can act as a broker and package cache for Salt clients.
+Registering clients on a {productname} Proxy is very similar to registering them directly on {productname}, with a few key differences.
 
+This section contains information on registering Salt clients on a Proxy using the {webui}, or with a bootstrap script.
```
Comment (keichwa) [inline modules/client-configuration/pages/client-proxy.adoc]:
```suggestion
This section contains information on registering Salt clients on a proxy using the {webui}, or with a bootstrap script.
```

File: modules/client-configuration/pages/client-proxy.adoc [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
+Within the {webui}, standard proxy pages will show information about both Salt and traditional clients.
 
+You can see a list of clients that are connected to a proxy by clicking on the name of the Proxy in menu:Main Navigation[Systems > Systems > Proxy], selecting the [guimenu]``Details`` tab, and then selecting the [guimenu]``Proxy`` tab.
```
Comment (keichwa) [inline modules/client-configuration/pages/client-proxy.adoc]:
```suggestion
You can see a list of clients that are connected to a proxy by clicking on the name of the proxy in menu:Main Navigation[Systems > Systems > Proxy], selecting the [guimenu]``Details`` tab, and then selecting the [guimenu]``Proxy`` tab.
```

File: modules/client-configuration/pages/client-proxy.adoc [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
+A list of chained proxies for a salt client can be seen by clicking on the name of the client in menu:Main Navigation[Systems > All], selecting the [guimenu]``Details`` tab, and then selecting the [guimenu]``Connection`` tab.
```
Comment (keichwa) [inline modules/client-configuration/pages/client-proxy.adoc]:
```suggestion
A list of chained proxies for a Salt client can be seen by clicking on the name of the client in menu:Main Navigation[Systems > All], selecting the [guimenu]``Details`` tab, and then selecting the [guimenu]``Connection`` tab.
```

File: modules/client-configuration/pages/clients-and-activation-keys.adoc [hunk 4]
```diff
@@ -124,8 +156,6 @@ Create the initial bootstrap script template from the command line on the {produ
 
 This command will generate the bootstrap script and place them in [path]``/srv/www/htdocs/pub/bootstrap``.
 
-Alternatively you may use the {webui} to create your bootstrap script template.
-For more information, see pass:c[xref:FILENAME.adoc#s3-sattools-config-bootstrap[]].
+Alternatively you can use the {productname}{webui} to create your bootstrap script template, from menu:Overview[Tasks].
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-and-activation-keys.adoc]:
```suggestion
Alternatively, you can use the {productname} {webui} to create your bootstrap script template, from menu:Overview[Tasks].
```

File: modules/client-configuration/pages/non-suse-clients.adoc [hunk 5]
```diff
@@ -0,0 +1,6 @@
+[[non-suse-clients]]
+= Other Clients
+
+It is possible to register clients using operating systems from Red Hat, CentOS, or Ubuntu.
+
+This section contains information specific to clients running operating systems other than those provided by {suse}.
```
Comment (keichwa) [inline modules/client-configuration/pages/non-suse-clients.adoc]: Wondering whether this also applies for opensuse?

File: modules/client-configuration/pages/salt-client-proxy.adoc [hunk 6]
```diff
@@ -1,60 +1,62 @@
...
 
-Registering Salt clients via {susemgrproxy} from the {webui}
-is done almost the same way as registering clients directly with the {productname} server.
-The difference is that you specify the name of the proxy in the [guimenu]``Proxy`` drop-box on the menu:Main Menu[Systems > Bootstrapping] page.
+// TODO: Might need an 'unsupported' note? LKB 2019-05-01
 
-.Bootstrapping a Salt Client With a Proxy
-image::proxy-saltbootstrap.png[scaledwidth=80%]
+Instead of the {webui}, you may use the command line to register a minion through a proxy.
+Note that this procedure requires that you have installed the salt package on the minion before registration, and have the Advanced systems module activated.
```
Comment (keichwa) [inline modules/client-configuration/pages/salt-client-proxy.adoc]:
```suggestion
This procedure requires that you have installed the Salt package on the minion before registration, and have the Advanced systems module activated.
```

File: modules/client-configuration/pages/script-client-proxy.adoc [hunk 7]
```diff
@@ -1,30 +1,28 @@
...
+. Optional: edit the resulting bootstrap script.
+. Execute the bootstrap script on the clients.
 
-. Optionally edit the resulting bootstrap script.
-Execute the bootstrap script on the clients as described in pass:c[xref:FILENAME.adoc#connect.first.client[]].
-
-
-The clients are registered with the {susemgrproxy} specified in the bootstrap script.
+// Not sure where this is pointing. LKB 2019-05-01
+// as described in pass:c[xref:FILENAME.adoc#connect.first.client[]].
```
Comment (keichwa) [inline modules/client-configuration/pages/script-client-proxy.adoc]: IIRC, this was a pointer to the Getting Started.  Grepping in maintenance/3.2...

```
doc-susemanager> grep connect.first.client adoc/*c
...
adoc/quickstart3_chap_suma_keys_and_first_client.adoc:[[connect.first.client]]
```

File: modules/client-configuration/pages/clients-and-activation-keys.adoc [hunk 8]
```diff
@@ -30,31 +17,76 @@ An activation key can specify:
...
+If you are using bootstrap scripts, consider creating an activation key for each script.
+This will help you align channel assignments, package installation, system group memberships, and configuration channel assignments.
+You will also need less manual interaction with your system after registration.
 
+.Bandwidth Requirements
 
-=== Activation Keys Best Practices
+Using activation keys might result in automatic downloading of software at registration time, which might not be desirable in environments where bandwidth is constrained.
 
+These events create bandwidth usage:
```
Comment (moio) [inline modules/client-configuration/pages/clients-and-activation-keys.adoc]: I would not call those "events", as they are really Activation Key configuration options.

File: modules/client-configuration/pages/non-suse-clients.adoc (hunk 5 above)
Comment (Loquacity) [reply modules/client-configuration/pages/non-suse-clients.adoc]: I would include opensuse as a suse OS.

**Comment by Loquacity** [comment]
> I'd always capitalize "Salt". I'm unsure about proxy. Maybe, capitalizing only if we talk about the SUMA Proxy product. There are corner cases, though.

Yeah, I'm not sure about capitalising salt, since we use it more as a general noun than a proper one (we don't capitalise "traditional", for example). That said, I don't mind which way we go, and it's probably capitalised  more than it isn't at this stage, so I'm happy to go with the caps.

**Comment by moio** [comment]
Well, Salt is many things.

Salt the company is actually called SaltStack Inc.

Salt the open source project ~~is called Salt Open~~ [used to be called Salt Open](https://web.archive.org/web/20160321012048/http://saltstack.com/community/) and [now is called just Salt](https://web.archive.org/web/20190427002321/https://www.saltstack.com/resources/community/). Note that [once upon a time, the name was actually SaltsStack](https://web.archive.org/web/20140302005638/http://saltstack.com/community/), so it has been renamed at least three times to this day.

`salt` the commandline tool is not capitalized and fortunately never really changed - the same holds for `salt-master` and `salt-minion` the services.

I call a minion a Salt minion, to distinguish it from Despicable Me characters. [Wikipedia has even more minion definitions](https://en.wikipedia.org/wiki/Minion) I was not aware of. Basically, unless we are really talking about code/code artifacts that really are uncapitalized and would not work otherwise, I do not see cases in which the name should not be capitalized.

**Comment by Loquacity** [comment]
> Well, Salt is many things.
> 
... snip
> I call a minion a Salt minion, to distinguish it from Despicable Me characters. [Wikipedia has even more minion definitions](https://en.wikipedia.org/wiki/Minion) I was not aware of. Basically, unless we are really talking about code/code artifacts that really are uncapitalized and would not work otherwise, I do not see cases in which the name should not be capitalized.

I've been using client rather than minion, mostly because I don't think minion would translate well across cultures (I've also got a personal story about use of the word that I'll share with you over a beer one day). Regarding caps, like I said before, I'm not particularly fussed either way. Generally speaking, we would only give initcaps to a proper noun, but the German influence has certainly crept into these  docs and there's lots of initcaps floating around, so I don't think it matters too much in the grand scheme of things. Happy to keep them!

**Comment by moio** [comment]
Hmm, we (developers) have been using "clients" as a shorthand for "traditional clients" sometimes. I try my best to always qualify them as "traditional", but I am pretty sure we have not been super-consistent with this one.

From a Spacewalk terminology POV, "systems" would probably be better, although it does not convey any architectural meaning, so it is not always appropriate.

Just keep in mind that in the UI/API/any other public-facing, code-determined place we basically always use "minions" (and I would say "Salt minions" would be even better) - to keep coherence with SaltStack Inc. (and open source community) use of the term.

As an engineer, my opinion is that once a word has been "borrowed" and enters some technical jargon for a given discipline, it should be used according to the jargon's definition only - any original meaning should be ignored as it is potentially misleading. We have many examples with varying degrees of cultural acceptance ranging from [master/slave buses](https://en.wikipedia.org/wiki/Master/slave_(technology) which are not about slavery, to [motherboards](https://en.wikipedia.org/wiki/Motherboard) which do not refer to parenthood, to [right angles](https://en.wikipedia.org/wiki/Right_angle) which are not about correctness.

I am also aware that is not universally accepted and some people continue insist with attach original meanings to words - sometimes causing heated debates. My suggestion would still be to include a dictionary of definitions and then stick to it.

**Comment by Loquacity** [comment]
> As an engineer, my opinion is that once a word has been "borrowed" and enters some technical jargon for a given discipline, it should be used according to the jargon's definition only - any original meaning should be ignored as it is potentially misleading. 

As a writer, I have been arguing that engineers should just stop borrowing words (I still hate 'instantiate' with a totally undeserved passion) ;)  

> I am also aware that is not universally accepted and some people continue insist with attach original meanings to words - sometimes causing heated debates. My suggestion would still be to include a dictionary of definitions and then stick to it.

Sadly, definition lists are rarely looked at by readers *or* writers, so their function is mostly decorative. Instead, I will sing the song of my people: Do whatever you like, as long as it's consistent.

**Pull request merged**